### ISSUE TITLE
Fix narrowing conversion warning.

### ISSUE THREAD

**Issue by racardoso** [PPC64/hhvm#110]
Compiling gives the following warning:

```
vasm-copy.cpp: In function ‘void HPHP::jit::{anonymous}::analyze_lea(HPHP::jit::{anonymous}::Env&, const HPHP::jit::lea&)’:
vasm-copy.cpp:247:42: warning: narrowing conversion of ‘lea.HPHP::jit::lea::s.HPHP::jit::Vptr::disp’ from ‘const intptr_t {aka const long int}’ to ‘int32_t {aka int}’ inside { } [-Wnarrowing]
dst = RegInfo { lea.s.base, lea.s.disp };
                                     ^
```

**Comment by racardoso** [comment]
Note that `struct RegInfo { Vreg base; int32_t disp; };`

**Comment by racardoso** [comment]
Fixed on 83a9c1ee253c0bf90b6af9bec0d47f7986331274 by @gut
